app/pages/settings.cjsx

```diff
@@ -9,6 +9,90 @@ apiClient = require '../api/client'
 putFile = require '../lib/put-file'
 
 MAX_AVATAR_SIZE = 65536
+MIN_PASSWORD_LENGTH = 8
+
+ChangePasswordForm = React.createClass
+  displayName: 'ChangePasswordForm'
+
+  getDefaultProps: ->
+    user: {}
+
+  getInitialState: ->
+    old: ''
+    new: ''
+    confirmation: ''
+    inProgress: false
+    success: false
+    error: null
+
+  render: ->
+    <form ref="form" onSubmit={@handleSubmit}>
+      <p>
+        <strong>Change your password</strong>
+      </p>
+
+      <table className="standard-table">
+        <tbody>
+          <tr>
+            <td>Current password</td>
+            <td><input type="password" className="standard-input" size="20" onChange={(e) => @setState old: e.target.value} /></td>
+          </tr>
+          <tr>
+            <td>New password</td>
+            <td>
+              <input type="password" className="standard-input" size="20" onChange={(e) => @setState new: e.target.value} />
+              {if @state.new.length isnt 0 and @tooShort()
+                <small className="form-help error">That’s too short</small>}
+            </td>
+          </tr>
+          <tr>
+            <td>Confirm new password</td>
+            <td>
+              <input type="password" className="standard-input" size="20" onChange={(e) => @setState confirmation: e.target.value} />
+              {if @state.confirmation.length >= @state.new.length - 1 and @doesntMatch()
+                <small className="form-help error">These don’t match</small>}
+            </td>
+          </tr>
+        </tbody>
+      </table>
+
+      <p>
+        <button type="submit" className="standard-button" disabled={not @state.old or not @state.new or @tooShort() or @doesntMatch() or @state.inProgress}>Change</button>{' '}
+
+        {if @state.inProgress
+          <i className="fa fa-spinner fa-spin form-help"></i>
+        else if @state.success
```

Comment:
I'd actually like to leave this. The real issue is that the auth modules needs to sign out and back in. There might be a way around that, but I don't think it's worth investing too much time in right now.